### DISCUSSION TITLE
Discuss Ansible roles and playbooks for automating Hub management tasks

### DISCUSSION THREAD

**Discussion by joewxboy** [open-horizon/devops#143] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by joewxboy** [comment]
One task would be to install the Mgmt Hub.  A second would be to upgrade to a newer version.

A second task would be to add an organization, and then a user in that org.

**Reply by loudonlune** [reply]
Starting with installing the management hub, I think a first good step would be to wrap the existing script and install its dependencies automatically.

In a local workspace, I've started developing an Ansible role that fulfills this task on a host.

Ansible has its own variable scope, which you can think of as a map with String keys and values of any type.
You can provide variables by passing them in on the command line, or by providing a YAML file via the command line.
The way to tell the management hub role how to install the management hub is to write up a YAML file, and then passing it into Ansible. However, in order to keep compatibility with the existing script, there is an option to provide a ".env" style configuration as well.

Since this is a containerized application, we should be able to control versioning of what is installed via the container tag.
Upgrading the version would be as simple as running the role again, but updating the config to have different versions.
Maybe we could also have a "build from source" option for the version, where it builds the container images from the main branches.

For adding and removing organizations/users inside organizations will be quite a bit different.
I think both of these tasks could be accomplished with one Ansible module.
It takes a minimum of three parameters, the organization to add, the user to add, and then whether the user should be "present" or "absent". If set to "present", then it creates the user... if "absent", then it deletes the user. For organizations, it will create it if not present, and then if there are no users remaining in the org, it will remove it.
